test/groovy/util/JenkinsPropertiesRule.groovy

```diff
@@ -38,10 +38,18 @@ class JenkinsPropertiesRule implements TestRule {
 
                 testInstance.helper.registerAllowedMethod("readProperties", [Map.class], {
                     propertyPath ->
-                        if (JenkinsPropertiesRule.this.propertyPath.contains(propertyPath.file)) {
+                        if(propertyPath.file){
+                            if (JenkinsPropertiesRule.this.propertyPath.contains(propertyPath.file)) {
                             return JenkinsPropertiesRule.this.configProperties
                         }
-
+                    }
+                        else if(propertyPath.text){
+                            def readPropertiesMap = [:]
+                            for (property in propertyPath) {
```

Comment:
This will not work. Here we use the `propertyPath` which is actually the map of arguments handed over to `readProperites`. This map is - for our tests - expected to hold either `text` or `file`. Me need to parse the `text` property here in this case which is expected to be a string instead of the map "as it".

Reply:
Sure, adapted.